node/cn/handler_msg_test.go

```diff
@@ -62,6 +63,146 @@ func prepareTestHandleNewBlockMsg(t *testing.T, mockCtrl *gomock.Controller, blo
 	return newBlock, msg, mockPeer, mockFetcher
 }
 
+func prepareDownloader(t *testing.T) (*gomock.Controller, *mocks2.MockProtocolManagerDownloader, *MockPeer, *ProtocolManager) {
+	mockCtrl := gomock.NewController(t)
+	mockDownloader := mocks2.NewMockProtocolManagerDownloader(mockCtrl)
+
+	mockPeer := NewMockPeer(mockCtrl)
+	mockPeer.EXPECT().GetID().Return(nodeids[0].String()).AnyTimes()
+
+	pm := &ProtocolManager{downloader: mockDownloader}
+
+	return mockCtrl, mockDownloader, mockPeer, pm
+}
+
+func TestHandleBlockHHeadersMsg(t *testing.T) {
+	headers := []*types.Header{blocks[0].Header(), blocks[1].Header()}
+	{
+		mockCtrl, _, mockPeer, pm := prepareDownloader(t)
+		msg := generateMsg(t, BlockHeadersMsg, blocks[0].Header())
+
+		assert.Error(t, handleBlockHeadersMsg(pm, mockPeer, msg))
```

Comment:
You didn't set the `EXPECT().DeliverHeaders`. In this case, it will return nil??
```
func handleBlockHeadersMsg(pm *ProtocolManager, p Peer, msg p2p.Msg) error {
	// A batch of headers arrived to one of our previous requests
	var headers []*types.Header
	if err := msg.Decode(&headers); err != nil {
		return errResp(ErrDecode, "msg %v: %v", msg, err)
	}
	if err := pm.downloader.DeliverHeaders(p.GetID(), headers); err != nil {
		logger.Debug("Failed to deliver headers", "err", err)
	}
	return nil
}
```

Reply:
Above test is to check if decoding error happens for wrong type of payload. It is `*types.Header`, not `[]*types.Header`. Therefore, it does not go to `DeliverHeader` but return error right after `Decode` fails.